code/cycler/src/wattrex_battery_cycler/app/app_man/app_man_core.py

```diff
@@ -0,0 +1,290 @@
+        self.raised_alarms: List[CyclerDataAlarmC] = []
+        ## Channel attributes to receive and send info
+        self.__chan_str_alarms = str_alarms #pylint: disable=unused-private-member
+        self.__chan_str_reqs = str_reqs
+        self.__chan_str_data = str_data
+        self.__cs_deprecated: bool = False
+        self.__wait_exp_reqst: bool = False
+        self.__wait_cs_reqst: bool = False
+        self.__iter: int = 0
+        ## Power control object
+        self.pwr_control: MidPwrControlC= MidPwrControlC(devices= devices,
+                            alarm_callback= self.alarm_callback, battery_limits=None,
+                            instruction_set=None)
+    @property
+    def gen_meas(self) -> CyclerDataGenMeasC|None:
+        """Return the local general measurements
+
+        Returns:
+            CyclerDataGenMeasC: General measurements
+        """
+        return self.__local_gen_meas
+
+    @property
+    def ext_meas(self) -> CyclerDataExtMeasC|None:
+        """Return the local external measurements
+
+        Returns:
+            CyclerDataExtMeasC: External measurements
+        """
+        return self.__local_ext_meas
+
+    @property
+    def all_status(self) -> CyclerDataAllStatusC|None:
+        """Return the local all status
+
+        Returns:
+            CyclerDataAllStatusC: All status
+        """
+        return self.__local_all_status
+
+    @property
+    def is_deprecated(self) -> bool:
+        """Return True if the cycler station is deprecated
+
+        Returns:
+            bool: True if the cycler station is deprecated
+        """
+        return self.__cs_deprecated
+
+
+    def update_local_data(self, new_gen_meas: CyclerDataGenMeasC, new_ext_meas: CyclerDataExtMeasC,
+                            new_all_status: CyclerDataAllStatusC,
+                            new_alarms: List[CyclerDataAlarmC]) -> None:
+        """Update the local data"""
+        self.raised_alarms.extend(new_alarms)
+        self.__local_all_status = new_all_status
+        self.__local_ext_meas = new_ext_meas
+        self.__local_gen_meas = new_gen_meas
+        self.pwr_control.update_local_data(self.__local_gen_meas, self.__local_all_status)
+
+    def process_recv_data(self):
+        """Process the requested made to the str node, 2 messages per iteration
+        """
+        log.debug("Processing request")
+        for _ in range(2):
+            msg: MidStrCmdDataC = self.__chan_str_data.receive_data_unblocking()
+            if msg is not None:
+                if msg.error_flag:
+                    log.debug(f"Error in the message containing {msg.cmd_type}")
+                    if (msg.cmd_type == MidStrDataCmdE.EXP_DATA and
+                        any(var is None for var in (msg.battery, msg.profile))):
+                        ## The experiment will be set to error
+                        self.__update_exp_status(exp_status=CyclerDataExpStatusE.ERROR)
```

Comment:
You can update the status of an unknown experiment? I think this is the case for the error flog on EXP_STATUS

Reply:
You know the experiment but no the battery and the profile